pkg/castemplate/v1alpha2/config.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2017 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	"github.com/openebs/maya/pkg/util"
+)
+
+// ConfigInterface abstracts get of various fields of config
+type ConfigInterface interface {
+	GetName() string
+	GetValue() string
+	IsEnabled() string
+	GetData() map[string]string
+}
+
+// EngineConfig contains highPriority and lowPriority config
+type EngineConfig struct {
```

Comment:
Engine does not need to support high & low priority.
Engine just provided a mechanism to merge two configs. While merging it was felt necessary to have one as high priority & other as a low priority.
A simple merge function is sufficient.
This merge function could also have been at `pkg/apis/openebs.io/v1alpha/cas_template.go` as well.
e.g. `MergeCasConfig(high, low []Config) []Config`

Reply:
Most of the use cases maynot need to merge configs.
Typical use cases will require a `config` & `runtimevalues`
IMO calling `runtimevalues` is better than `runtimeconfig`.
Both of these represent different aspects